radix-engine/src/blueprints/resource/resource_manager.rs

```diff
@@ -988,19 +970,18 @@ impl ResourceManagerBlueprint {
                 }
             }
 
-            let node_id = api.kernel_allocate_node_id(RENodeType::Bucket)?;
-            api.kernel_create_node(
-                node_id,
-                RENodeInit::NonFungibleBucket(
-                    BucketInfoSubstate {
-                        resource_address,
-                        resource_type: ResourceType::NonFungible { id_type },
-                    },
-                    LiquidNonFungibleResource::new(ids.clone()),
+            let info = BucketInfoSubstate {
+                resource_address,
+                resource_type: ResourceType::NonFungible { id_type },
+            };
+            let bucket_id = api.new_object(
+                BUCKET_BLUEPRINT,
+                btreemap!(
+                    0 => scrypto_encode(&info).unwrap(),
+                    1 => scrypto_encode(&LiquidNonFungibleResource::new(ids.clone())).unwrap()
```

Comment:
Offset `1` is used for both fungible and non-fungible resource. 

Before future refactoring and engine does not consider application-level type info (fungibility, divisibility and local id type), can we please have the following schema for buckets and vaults?
```
0 - info
1 - liquid fungible
2 - locked fungible
3 - liquid non-fungible
4 - locked non-fungible
```

Would also like to have named constants for all substate offsets.

Reply:
Based on discussion, we won't go with this approach and will use vector (tuple) of substates rather a map.